DQM/SiPixelPhase1Common/src/SiPixelCoordinates.cc

```diff
@@ -0,0 +1,916 @@
+// -*- C++ -*-
```

Comment:
Re where to put this: Originally, this file is by @jkarancs and lives here: https://github.com/jkarancs/PhaseIPixelNtuplizer/blob/31e12769af7eb82dd7c1f1c4418641e4d2fe8b7d/src/SiPixelCoordinates.cc
There were some plans to put this into a DPGAnalysis package (to be created), but I think having DQM depend on that is not good either. So, we are looking for a better place, but to have it available now I just threw it into the DQM package. If you have better suggestions (this is not Phase1, but Pixel specific; it is for now only used in DQM and that will probably stay like that; but external analysis code will also use it) please let us know.

Re the caching: Yes, this is probably too aggressive. There are some really slow operations in there (around channels/ROCs), but in many cases avoiding it might be better. From what I have measured, this class causes ~10% of the *online DQM* cost (which is basically only Pixel DQM), in usual offline sequences the Pixel DQM is only a few percent of the total. So there is an opportunity to improve a bit, but it seems not too urgent to me and I wanted to keep @jkarancs's state for now (up to a bugfix).